### PR TITLE
Rg 3

### PR DESCRIPTION
**Pull request by AntonFartovii**
Main layout for pages. With
- Sidebar,
- Header,
- children,
- Footer.

`next dev`, `npm run build`, `npm run start`- works correctly with layout

image

### CODE REVIEW

**Review by ImmelstronDev — APPROVED** [review]

**Review by ImmelstronDev — APPROVED** [review]

File: src/app/components/MainLayout.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import Header from "@/app/components/Header";
+import React, {ReactNode} from 'react';
+import Sidebar from "@/app/components/Sidebar";
+import Footer from "@/app/components/Footer";
+
+interface LayoutProps {
+  children: ReactNode;
+}
+
+const MainLayout = ({children}: LayoutProps) => {
```
Comment (sekaa4) [inline src/app/components/MainLayout.tsx]: You can write type PropsWithChildren(this type define in react), if you use only children, or if you want use children and another props you can use PropsWithChildren\<interface>\ and then as example 
const {children, otherprops} = props

**Review by sekaa4 — CHANGES_REQUESTED** [review]
Please watch structure for the project here https://github.com/devianllert/nextplate, folder src/widgets
/layouts/ should contain your layout components

File: src/app/components/Footer.tsx [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
```
Comment (ImmelstronDev) [inline src/app/components/Footer.tsx]: 1) React import is not needed, NEXT under the hood works with jsx

File: src/app/components/Header.tsx [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
```
Comment (ImmelstronDev) [inline src/app/components/Header.tsx]: The same as in point 1

File: src/app/components/MainLayout.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import Header from "@/app/components/Header";
+import React, {ReactNode} from 'react';
```
Comment (ImmelstronDev) [inline src/app/components/MainLayout.tsx]: The same as in point 1

File: src/app/components/Sidebar.tsx [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
```
Comment (ImmelstronDev) [inline src/app/components/Sidebar.tsx]: The same as in point 1

**Review by ImmelstronDev — CHANGES_REQUESTED** [review]

**Review by ImmelstronDev — APPROVED** [review]

**Review by sekaa4 — APPROVED** [review]

**Pull request merged**